component/discovery/consulagent/promtail_consulagent_test.go

```diff
@@ -33,6 +33,8 @@ import (
 
 //nolint:interfacer // this follows the pattern in prometheus service discovery
 func TestMain(m *testing.M) {
+	m.Run()
+	// Verify no leaks after tests run.
```

Comment:
the function VerifyTestMain is actually already calling m.Run():
```
func VerifyTestMain(m TestingM, options ...Option) {
	exitCode := m.Run()
	opts := buildOpts(options...)

	var cleanup func(int)
	cleanup, opts.cleanup = opts.cleanup, nil
	if cleanup == nil {
		cleanup = _osExit
	}
	defer func() { cleanup(exitCode) }()

	if exitCode == 0 {
		if err := Find(opts); err != nil {
			fmt.Fprintf(_osStderr, "goleak: Errors on successful test run: %v\n", err)
			exitCode = 1
		}
	}
}
```

Reply:
Right, I misread the `goleak` and the `TestMain` is kinda new to me ;)